harmonica/_equivalent_sources/cartesian.py

```diff
@@ -326,10 +318,8 @@ def predict(self, coordinates):
         check_is_fitted(self, ["coefs_"])
         shape = np.broadcast(*coordinates[:3]).shape
         size = np.broadcast(*coordinates[:3]).size
-        coordinates = tuple(
-            np.atleast_1d(i.astype(self.dtype)).ravel() for i in coordinates[:3]
-        )
-        data = np.zeros(size, dtype=self.dtype)
+        coordinates = tuple(np.atleast_1d(i).ravel() for i in coordinates[:3])
+        data = np.zeros(size)
```

Comment:
By default this array will be a float64, so let's make it more explicit.
```suggestion
        data = np.zeros(size, dtype=np.float64)
```